x/farming/types/keys.go

```diff
@@ -122,7 +113,20 @@ func GetRewardByFarmerAddrIndexPrefix(farmerAcc sdk.AccAddress) []byte {
 	return append(RewardByFarmerAddrIndexKeyPrefix, address.MustLengthPrefix(farmerAcc.Bytes())...)
 }
 
-func GetRewardStakingCoinDenomFromIndexKey(key []byte) string {
+// ParseRewardKey parses a RewardKey.
+func ParseRewardKey(key []byte) (stakingCoinDenom string, farmer sdk.AccAddress, err error) {
+	denomLen := key[1]
+	stakingCoinDenom = string(key[2 : 2+denomLen])
+	farmer = key[2+denomLen+1:]
+	// TODO: add error case
+	return
+}
+
+// ParseRewardByFarmerAddrIndexKey parses a key of RewardByFarmerAddrIndex from bytes.
+func ParseRewardByFarmerAddrIndexKey(key []byte) (stakingCoinDenom string, farmer sdk.AccAddress, err error) {
```

Comment:
How about the arguments ordering to `func ParseRewardByFarmerAddrIndexKey(key []byte) (farmer sdk.AccAddress, stakingCoinDenom string, err error)` to consistent with indexing ordering

Reply:
Ok, it makes sense.